internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -919,3 +952,367 @@ func TestMachineSetReconciler_updateStatusResizedCondition(t *testing.T) {
+				Namespace:   namespace.Name,
+				Labels:      map[string]string{},
+				Annotations: map[string]string{},
+				ManagedFields: []metav1.ManagedFieldsEntry{
+					{
+						Manager:    "manager",
+						Operation:  metav1.ManagedFieldsOperationUpdate,
+						FieldsType: "FieldsV1",
+						FieldsV1:   &metav1.FieldsV1{Raw: fieldV1},
+					},
+				},
+				Finalizers: []string{"testing-finalizer"},
+			},
+			Spec: clusterv1.MachineSpec{
+				ClusterName: testClusterName,
+				InfrastructureRef: corev1.ObjectReference{
+					Namespace: namespace.Name,
+				},
+				Bootstrap: clusterv1.Bootstrap{
+					DataSecretName: pointer.String("machine-bootstrap-secret"),
+				},
+			},
+		}
+		g.Expect(env.Create(ctx, deletingMachine)).To(Succeed())
+		// Delete the machine to put it in the deleting state
+		g.Expect(env.Delete(ctx, deletingMachine)).To(Succeed())
+		// Wait till the machine is marked for deletion
+		g.Eventually(func() bool {
+			if err := env.Get(ctx, client.ObjectKeyFromObject(deletingMachine), deletingMachine); err != nil {
+				return false
+			}
+			return !deletingMachine.DeletionTimestamp.IsZero()
+		}, timeout).Should(BeTrue())
+
+		machines := []*clusterv1.Machine{inPlaceMutatingMachine, deletingMachine}
+
+		// Sync the Machines.
+		reconciler := &Reconciler{Client: env}
+		g.Expect(reconciler.syncMachines(ctx, ms, machines)).To(Succeed())
+
+		// The in-place mutating machine should have:
+		// - clean-up managed fields
+		// - updated in-place propagating values
+		updatedInPlaceMutatingMachine := machines[0]
+		// Verify ManagedFields
+		g.Expect(updatedInPlaceMutatingMachine.ManagedFields).Should(
+			ContainElement(ssa.MatchManagedField(machineSetManagerName, metav1.ManagedFieldsOperationApply)),
+			"in-place mutable machine should contain an entry for SSA manager",
+		)
+		g.Expect(updatedInPlaceMutatingMachine.ManagedFields).ShouldNot(
+			ContainElement(ssa.MatchManagedField("manager", metav1.ManagedFieldsOperationUpdate)),
+			"in-place mutable machine should not contain an entry for old manager",
+		)
+		// Verify Labels
+		g.Expect(updatedInPlaceMutatingMachine.Labels).Should(HaveKeyWithValue("machine-set-matching-label", "true"))
+		// Verify Annotations
+		g.Expect(updatedInPlaceMutatingMachine.Annotations).Should(HaveKeyWithValue("annotation-1", "true"))
+		g.Expect(updatedInPlaceMutatingMachine.Annotations).Should(HaveKeyWithValue("precedence", "MachineSet"))
+		// Verify Node timeout values
+		g.Expect(updatedInPlaceMutatingMachine.Spec.NodeDrainTimeout).Should(And(
+			Not(BeNil()),
+			HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
+		))
+		g.Expect(updatedInPlaceMutatingMachine.Spec.NodeDeletionTimeout).Should(And(
+			Not(BeNil()),
+			HaveValue(Equal(*ms.Spec.Template.Spec.NodeDeletionTimeout)),
+		))
+		g.Expect(updatedInPlaceMutatingMachine.Spec.NodeVolumeDetachTimeout).Should(And(
+			Not(BeNil()),
+			HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
+		))
```

Comment:
```suggestion
		// Wait until Machine is up-to-date in cache.
		g.Eventually(func(g Gomega) {
			// The in-place mutating machine should have:
			// - clean-up managed fields
			// - updated in-place propagating values
			updatedInPlaceMutatingMachine := inPlaceMutatingMachine.DeepCopy()
			g.Expect(env.Get(ctx, client.ObjectKeyFromObject(updatedInPlaceMutatingMachine), updatedInPlaceMutatingMachine))

			// Verify ManagedFields
			g.Expect(updatedInPlaceMutatingMachine.ManagedFields).Should(
				ContainElement(ssa.MatchManagedField(machineSetManagerName, metav1.ManagedFieldsOperationApply)),
				"in-place mutable machine should contain an entry for SSA manager",
			)
			g.Expect(updatedInPlaceMutatingMachine.ManagedFields).ShouldNot(
				ContainElement(ssa.MatchManagedField("manager", metav1.ManagedFieldsOperationUpdate)),
				"in-place mutable machine should not contain an entry for old manager",
			)

			// Verify in-place mutable fields have been updated.
			// Verify Labels
			g.Expect(updatedInPlaceMutatingMachine.Labels).Should(HaveKeyWithValue("machine-set-matching-label", "true"))
			// Verify Annotations
			g.Expect(updatedInPlaceMutatingMachine.Annotations).Should(HaveKeyWithValue("annotation-1", "true"))
			g.Expect(updatedInPlaceMutatingMachine.Annotations).Should(HaveKeyWithValue("precedence", "MachineSet"))
			// Verify Node timeout values
			g.Expect(updatedInPlaceMutatingMachine.Spec.NodeDrainTimeout).Should(And(
				Not(BeNil()),
				HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
			))
			g.Expect(updatedInPlaceMutatingMachine.Spec.NodeDeletionTimeout).Should(And(
				Not(BeNil()),
				HaveValue(Equal(*ms.Spec.Template.Spec.NodeDeletionTimeout)),
			))
			g.Expect(updatedInPlaceMutatingMachine.Spec.NodeVolumeDetachTimeout).Should(And(
				Not(BeNil()),
				HaveValue(Equal(*ms.Spec.Template.Spec.NodeDrainTimeout)),
			))
		}, 5*time.Second).Should(Succeed())
```

Let's please not rely on the local objects that CR just updated. Let's get the Machine from the testenv

Reply:
Switching to APIReader to solve the cache problem.